### PR TITLE
Fix TypeError: tuple indices must be integers or slices, not str issue with collate_fn

### PR DESCRIPTION
**Pull request by tambulkar**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [x] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.
https://github.com/pytorch/torchtune/issues/1010

#### Changelog
What are the changes made in this PR?
Change indexing

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help.)

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add unit tests for any new functionality
- [ ] update docstrings for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
	- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

### CODE REVIEW

**Comment by RdoubleA** [comment]
Appreciate you patching this. Actually we should go the other way - we are standardizing on outputting dictionaries for each sample now instead of a tuple of list[int]. `TextCompletionDataset` should be updated to return a dictionary.

Since you're already here you can do this or I can make the change, let me know how you want to proceed :)

**Comment by tambulkar** [comment]
I can go ahead and amend this PR - give me a few minutes

**Comment by RdoubleA** [comment]
you'll need to update `test_text_completion_dataset.py`, `test_wikitext_dataset.py`, and `test_cnn_dailymail_dataset.py` as well. You can run these tests locally with `pytest tests/torchtune/datasets`.

File: torchtune/utils/collate.py [hunk 1]
```diff
@@ -30,8 +30,8 @@ def padded_collate(
 
     Example:
         >>> token_pairs = [
-        >>>    ([1, 2, 3], [4, 5, 6]),
-        >>>    ([7,], [10,],),
+        >>>    {"tokens": [1, 2, 3], "labels": [4, 5, 6]},
```
Comment (RdoubleA) [inline torchtune/utils/collate.py:33]: awesome catch!

File: torchtune/datasets/_text_completion.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def __getitem__(self, index: int) -> Tuple[List[int], List[int]]:
         sample = self._data[index]
         return self._prepare_sample(sample)
 
-    def _prepare_sample(self, sample: Mapping[str, Any]) -> Tuple[List[int], List[int]]:
+    def _prepare_sample(self, sample: Mapping[str, Any]) -> Dict[str, List[int]]:
```
Comment (RdoubleA) [inline torchtune/datasets/_text_completion.py:52]: you'll have to update the return type of the above function as well. running `pre-commit` should catch these

**Review by RdoubleA — APPROVED** [review]
so glad you caught this early, thanks for jumping on a fix! were you able to launch your run successfully?

**Comment by RdoubleA** [comment]
Confirmed it runs on my end with the changes by running one of the built-in configs with wikitext

**Pull request merged**